drivers/firmware/scmi/core.c

```diff
@@ -15,6 +15,8 @@ LOG_MODULE_REGISTER(scmi_core);
 #define SCMI_CHAN_LOCK_TIMEOUT_USEC 500
 #define SCMI_CHAN_SEM_TIMEOUT_USEC 500
 
+static sys_slist_t scmi_event_list;
```

Comment:
Do we really need a linked list? Can we not have each protocol keep an array of `struct scmi_protocol_event` for all registered events?

The definition of `struct scmi_protocol_event` would be:

```
struct scmi_protocol_event {
    uint32_t id; /* event ID */
    scmi_evt_callback_t cb; /* event callback */
};
```